src/main/java/org/elasticsearch/index/analysis/SkroutzGreekStemmer.java

```diff
@@ -1079,49 +1049,43 @@ private boolean endsWithVowelNoY(char s[], int len) {
   }
 
   /**
-   * Creates a CharArraySet from a file.
-   *
-   * @param stopwords
-   *          Input stream from the stopwords file
-   *
-   * @param matchVersion
-   *          the Lucene version for cross version compatibility
-   * @return a CharArraySet containing the distinct stopwords from the given
-   *         file
-   * @throws IOException
-   *           if loading the stopwords throws an {@link IOException}
-   */
-  private static CharArraySet loadStopwordSet(InputStream stopwords,
-      Version matchVersion) throws IOException {
-    Reader reader = null;
-    try {
-      reader = IOUtils.getDecodingReader(stopwords, IOUtils.CHARSET_UTF_8);
-      return WordlistLoader.getWordSet(reader, matchVersion);
-    } finally {
-      IOUtils.close(reader);
-    }
-  }
-
-  /**
    * Helper function loading an exception Map by its name from the config file.
    * @param exName The exceptions name
-   * @return a Map of "term" -> ['s','t','e','m','m','e','d']
+   * @return a CharArrayMap which maps terms to their exceptions
    * @throws FileNotFoundException
    */
-  private static Map<String, char[]> loadExceptions(String exName)
+  private CharArrayMap<char[]> loadExceptions(String exName)
           throws FileNotFoundException {
     InputStream in = SkroutzGreekStemmer.class.getClassLoader()
             .getResourceAsStream("stemmer.yml");
     Yaml reader = new Yaml();
 
     Map<String, Map<String, String>> composite = (Map) reader.load(in);
     Map<String, String> stemException = composite.get(exName);
-    Map<String, char[]> rv = new HashMap<String, char[]>();
+    Map<char[], char[]> rv = new HashMap<char[], char[]>();
```

Comment:
I am just suggesting that rv is converted into a CharArrayMap, rather than having and using a HashMap, which is then converted to a CharArrayMap. It would be simpler if it was a CharArrayMap from the beginning.